src/learning/ProfileInitializer.cpp

```diff
@@ -199,16 +201,25 @@ std::vector<Perf> ProfileInitializer::initializeProfilePerformance(
 }
 
 void ProfileInitializer::initializeProfiles(MRSortModel &model) {
-  std::vector<Performance> perf_vec;
-  Performance firstAltPerf = altPerformance_.getPerformanceTable()[0];
-  std::vector<std::string> criteriaIds = firstAltPerf.getCriterionIds();
-  std::vector<float> catFreq = ProfileInitializer::categoryFrequency();
-  for (std::string criterion : criteriaIds) {
-    // OPTIM : POSSIBILITY parallelization asynchrone
-    std::vector<Perf> p = ProfileInitializer::initializeProfilePerformance(
-        criterion, model.categories, catFreq);
-    perf_vec.push_back(Performance(p));
+  bool ordered = 0;
+  while (!ordered) {
+    std::vector<Performance> perf_vec;
+    Performance firstAltPerf = altPerformance_.getPerformanceTable()[0];
+    std::vector<std::string> criteriaIds = firstAltPerf.getCriterionIds();
+    std::vector<float> catFreq = ProfileInitializer::categoryFrequency();
+    // CatFreq has a problem
+    for (std::string criterion : criteriaIds) {
+      // OPTIM : POSSIBILITY parallelization asynchrone
+      std::vector<Perf> p = this->initializeProfilePerformance(
+          criterion, model.categories, catFreq);
+      perf_vec.push_back(Performance(p));
+    }
+    try {
+      Profiles p = Profiles(perf_vec, "crit");
+      model.profiles = p;
+      ordered = 1;
+    } catch (std::invalid_argument const &err) {
+      ordered = 0;
```

Comment:
nit: you do not need any instruction in the catch statement, as ordered is already at false by default